### ISSUE TITLE
Ability to set visiblity option in personnal meta-data fields

### ISSUE THREAD

**Issue by roipoussiere** [tootsuite/mastodon#8243]
Currently, meta-data are always public, even on private accounts.

However, the ability to set the visibility on each meta-data fields could be interesting for several reasons.

## Communicate private informations

As a user, I would like to share some more private informations, for example my birth date. 🎂
So if I could fill the field and set it to private, only my followers will see this.

## Avoid massive meta-data collection

It's actually possible to collect meta-data.

For example, I could write a script to get the list of users living on a specific city. 🏙️
This behavior could bother some people and could be avoided by private fields.

## Prevent targeted harassment

Private fields could prevent a user to follow an account according to its meta-data, with bad intentions in mind. 😈

For example, some people may inform their religion or sexual orientation in meta-data.

A malicious user could follow and contact them based on this information, in order to annoy them, which can be very inconvenient if coupled with the point above.

---

## Suggested fields visibility

- public (everyone)
- local (instance only)
- private (followers only)
- mutuals (followers and following), linked with #7135

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by joachimesque** [comment]
Yes. I'd also like to specify fields visibility the links on my profile—some links for everyone, some only for private eyes.
